internal/sync/lifecycle.go

```diff
@@ -115,12 +155,15 @@ func (l *LifecycleOnce) Stop(f func() error) error {
 		return l.stopErr
 	}
 
-	if f == nil {
-		f = func() error { return nil }
+	// Unblock WaitForStart callers (ccounting Stop() without Start())
+	if l.started != nil {
+		close(l.started)
```

Comment:
I believe that the lock and the only-once guards against multiple attempts to close.